compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -1506,4 +1506,63 @@ def IREELinalgExt_WinogradOutputTransformOp : IREELinalgExt_Op<"winograd.output_
 
 } // OpGroupWinogradOps
 
+//===---------------------------------------------------------------------===//
+// Custom tilable op
+//===---------------------------------------------------------------------===//
+
+def IREELinalgExt_CustomOp : IREELinalgExt_Op<"custom_op"> {
+  let summary = "Custom operation for compiling with IREE";
+  let description = [{
+    This operation is meant to allow computation sequences that are fused at
+    tile level prescriptively. This is to account for cases where such fusion
+    cannot/is not yet discovered appropriately.
+
+    The operation implements all the interfaces needed to be able to
+    1. Compile e2e using IREE
+    2. Still be able to fuse with other operations that the compiler can
+       figure out automatically.
+
+    The operation has operands similar to `LinalgOp`s
+    1. Indexing maps : Modification on top of Linalg Op is that the indexing
+       map can be `affine_map<() -> ()>` indicating that operand cannot
+       be fused along and that this operand is not tiled
+    2. Iterator types
+    3. Inputs
+    4. Outputs: Which are "tied" operands for the results (implements the
+       `DestinationStyleOpInterface`).
+
+    The region of the operation represents the computation at a tile granularity.
+    The basic block arguments correspond to tiles of inputs/outputs. The region
+    yields tiled results of the same type as the corresponding basic block
+    arguments for the outputs. 
```

Comment:
I think the block arguments could use more description. Are statically shaped block arguments allowed? I would assume no, as it would be unclear how to tile along the statically-casted tile dimensions.

Reply:
I updated the description. PTAL